logic/src/commonMain/kotlin/com/wire/kalium/logic/configuration/server/ServerConfig.kt

```diff
@@ -180,10 +206,15 @@ class ServerConfigMapperImpl(
             blackList = blackList,
             teams = teams,
             title = title,
-            isOnPremises = isOnPremises
+            isOnPremises = isOnPremises,
+            proxy = proxy?.let { ServerConfig.Proxy(it.isProxyNeedsAuthentication, it.apiProxy, it.proxyPort) }
```

Comment:
I see this proxy object mapped manually many times, I suggest to have a mapper for the Proxy object, then we only call it to prevent false mappings. in the end those prxyMapper functions can be tested.